pkg/operator/controller/instance_controller.go

```diff
@@ -543,18 +541,21 @@ func (r *InstanceReconciler) deleteFinalizer(ctx context.Context, instance *clus
 	return r.Update(ctx, instance)
 }
 
-// createOrUpdateResource uses for creates or updates k8s resource, in our case used only for deployments.
 func (r *InstanceReconciler) createOrUpdateResource(ctx context.Context, object client.Object) error {
-	// Use CreateOrUpdate to create or update the PVC for cl-controlplane
-	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, object, func() error {
-		createTime := object.GetCreationTimestamp()
-		if createTime.IsZero() {
-			r.Logger.Infof("Create resource %v Name: %s Namespace: %s", reflect.TypeOf(object), object.GetName(), object.GetNamespace())
-		}
+	key := client.ObjectKeyFromObject(object)
+	err := r.Client.Create(ctx, object)
+	if err == nil {
+		r.Logger.Infof("Create resource %s Name: %s Namespace: %s", reflect.TypeOf(object), object.GetName(), object.GetNamespace())
 		return nil
-	})
+	}
+
+	if errors.IsAlreadyExists(err) {
+		// If resource already exists, update it
+		err = r.Client.Update(ctx, object)
```

Comment:
is there a need to Get the resource first so that you only override specific fields (and also have the correct resource version)?
Not sure...